File: 0824-number-of-lines-to-write-string/0824-number-of-lines-to-write-string.java
class Solution {
    public int[] numberOfLines(int[] widths, String s) {
        int lines = 1, sum = 0;

        for (char c : s.toCharArray()) {
            int width = widths[c - 'a'];
            if (sum + width > 100) {
                lines++;
                sum = width; //  line starts with current character
            } else {
                sum += width;
            }
        }

        return new int[]{lines, sum};
    }
}

File: 0824-number-of-lines-to-write-string/0824-number-of-lines-to-write-string.cpp
;
class Solution {
public:
    vector<int> numberOfLines(vector<int>& widths, string s) {

        int sum = 0, line = 1;
        for (char c : s) {
            int width = widths[c - 'a'];

            if (sum + width > 100) {
                line++;
                sum = width;
            } else {
                sum += width;
            }
        }
        return {line, sum};
    }
};